### PR TITLE
Improved Tests for Debugging Helix Run

### PR DESCRIPTION
**Pull request by shmao**
Improved the test for #10843. The issue reproduces in helix run only, which make it hard to root cause the issue. Adding more debug info in the test so that we can get more info regarding to the helix test failure.

Improved the test for #10848. Added more debug info in the test.

Fixed an issue with System.Runtime.Serialization.* tests. PR #11480  extracted settings into settings.targets. The issue is that VS couldn't find source code files included in settings.targets.

### CODE REVIEW

**Comment by shmao** [comment]
@mconnew @huanwu @zhenlan can you please review the PR? Thanks!

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 1]
```diff
@@ -42,26 +42,51 @@ public static void XmlBaseWriter_WriteBase64Async()
...
-            writer.WriteEndElement();
-            writer.WriteEndDocument();
-            var task = writer.FlushAsync();
-            task.Wait();
-            ms.Position = 0;
-            var sr = new StreamReader(ms);
-            actual = sr.ReadToEnd();
+            messageBuilder.AppendLine($"An error occured: {e.Message}");
+            messageBuilder.AppendLine(e.StackTrace);
+            Assert.False(false, messageBuilder.ToString());
```
Comment (mconnew) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Assert.True

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 2]
```diff
@@ -42,26 +42,51 @@ public static void XmlBaseWriter_WriteBase64Async()
...
         string expect = GetExpectString(bytes, byteSize);
-        using (var ms = new AsyncMemoryStream())
+        var messageBuilder = new StringBuilder();
+        try
+        {            
+            using (var ms = new AsyncMemoryStream())
+            {
+                var writer = XmlDictionaryWriter.CreateTextWriter(ms);
+                messageBuilder.AppendLine("XmlDictionaryWriter.CreateTextWriter() completed.");
+
```
Comment (mconnew) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: No need to keep appending to a message builder. You can have a string and just change the value after each step.

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs [hunk 3]
```diff
@@ -1942,10 +1942,29 @@ public static void DCS_TypeWithCollectionWithoutDefaultConstructor()
...
+        try
         {
             serializer.ReadObject(new MemoryStream());
-        });
+        }
+        catch (Exception e)
+        {
+            Type expectedExceptionType = typeof(XmlException);
+            Type actualExceptionType = e.GetType();
+            if (actualExceptionType != expectedExceptionType)
```
Comment (mconnew) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: Use .Equals as Type has an implementation of .Equals and it's what xunit is using. Also Type.Equals uses Type.UnderlyingSystemType which is abstract and does a ReferenceEquals against those type objects. If the types at runtime are different (e.g. RTType vs some other implementation), you might get different implementations of Type which return something different from UnderlyingSystemType. If it's in contract, try outputting the type information for the objects returned from UnderlyingSystemType and see if they are the same.

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs (hunk 1 above)
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Changed.

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs (hunk 2 above)
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Nice suggestion. Changed.

File: src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs (hunk 3 above)
Comment (shmao) [inline src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs]: Good point. Changed.

File: src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs [hunk 4]
```diff
@@ -42,26 +42,54 @@ public static void XmlBaseWriter_WriteBase64Async()
...
+            var sb = new StringBuilder();
+            sb.AppendLine($"An error occured: {e.Message}");
+            sb.AppendLine(e.StackTrace);
+            sb.AppendLine();
+            sb.AppendLine($"The last completed operation before the exception was: {lastCompletedOperation}");
+            Assert.True(false, sb.ToString());
         }
 
         Assert.StrictEqual(expect, actual);
+
```
Comment (mconnew) [inline src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs]: Nit:empty line

**Pull request merged**